bftengine/src/bcstatetransfer/DBDataStore.cpp

```diff
@@ -256,40 +232,30 @@ void DBDataStore::deserializeResPage(
   Serializable::deserialize(is, outPage, sizeOfReseredPage);
 }
 void DBDataStore::loadResPages() {
-  LOG_DEBUG(logger(), "");
-  for (uint32_t pageid = 0; pageid < inmem_->getNumberOfReservedPages(); ++pageid) {
-    for (uint64_t chkp = 1; chkp <= inmem_->getMaxNumOfStoredCheckpoints(); ++chkp) {
-      Sliver dynamic_key;
-      if (!get(staticResPageKey(pageid, chkp), dynamic_key)) continue;
-      LOG_DEBUG(logger(), "[" << pageid << ", " << chkp << "] => " << dynamic_key);
-      Sliver serializedPage;
-      if (!get(dynamic_key, serializedPage)) {
-        LOG_ERROR(logger(), "failed to load page for [" << staticResPageKey(pageid, chkp) << ":" << dynamic_key << "]");
-        continue;
-      }
-      std::istringstream iss(
-          std::string(reinterpret_cast<const char*>(serializedPage.data()), serializedPage.length()));
-      uint32_t pageid;
-      uint64_t checkpoint;
-      STDigest digest;
-      char* page = new char[inmem_->getSizeOfReservedPage()];
-      deserializeResPage(iss, pageid, checkpoint, digest, page);
-      inmem_->setResPage(pageid, checkpoint, digest, page);
-      delete[] page;  // is copied
-    }
+  auto it(dbc_->getIteratorGuard());
+  for (auto keyValue = it->seekAtLeast(dynamicResPageKey(0, 0));  // start of dynamic keys space
+       !it->isEnd() && keyValue.first.string_view().find(keymanip_->getReservedPageKeyPrefix().string_view()) == 0;
+       keyValue = it->next()) {
+    std::istringstream iss(
+        std::string(reinterpret_cast<const char*>(keyValue.second.data()), keyValue.second.length()));
+    uint32_t pageid;
+    uint64_t checkpoint;
+    STDigest digest;
+    char* page = new char[inmem_->getSizeOfReservedPage()];
+    deserializeResPage(iss, pageid, checkpoint, digest, page);
+    inmem_->setResPage(pageid, checkpoint, digest, page);
+    delete[] page;  // is copied
+    LOG_DEBUG(logger(), KVLOG(pageid, checkpoint, digest.toString()));
```

Comment:
I would add a comment describing the algorithm

Reply:
sure